qt/widgets/common/src/MuonFitPropertyBrowser.cpp

```diff
@@ -1483,12 +1499,110 @@ void MuonFitPropertyBrowser::clearChosenPeriods() const {
+  // this is for switching but has a bug at the moment
+  // const QStringList &selected) {
+  if (m_periodsToFitOptions[j] == CUSTOM_LABEL) {
+    // currently the below does not work reliably (if period arithmatic is
+    // presemnt it gives bad results
+    /*
+    m_enumManager->setEnumNames(m_periodsToFit, m_periodsToFitOptions);
+    m_enumManager->setValue(m_periodsToFit, j);
+    setChosenPeriods(selected);*/
+    // lets default to all periods for now
+    m_enumManager->setEnumNames(m_periodsToFit, m_periodsToFitOptions);
+    m_enumManager->setValue(m_periodsToFit, j);
+    // explictly set all periods
+    setAllPeriods();
+  } else if (m_periodsToFitOptions[j] == ALL_PERIODS_LABEL) {
+    m_enumManager->setEnumNames(m_periodsToFit, m_periodsToFitOptions);
+    m_enumManager->setValue(m_periodsToFit, j);
+    // explictly set all periods
+    setAllPeriods();
+  } else { // single number
+    m_enumManager->setEnumNames(m_periodsToFit, m_periodsToFitOptions);
+    m_enumManager->setValue(m_periodsToFit, j);
+    setChosenPeriods(m_periodsToFitOptions[j]);
+  }
+}
+/**
+* Adds a new checkbox to the list of periods with given name
+* It updates the display
+* @param name :: [input] Name of period to add
+*/
+void MuonFitPropertyBrowser::addPeriodCheckboxToMap(const QString &name) {
+  if (m_periodBoxes.find(name) != m_periodBoxes.end()) {
+    // if the box already exists
+    return;
+  }
+  // has to go here to get the original value
+  int j = m_enumManager->value(m_periodsToFit);
+  // auto selected = getChosenPeriods();
+  addPeriodCheckbox(name);
+  updatePeriods(j);
+}
+
 /**
 * Add a new checkbox to the list of periods with given name
 * The new checkbox is unchecked by default
 * @param name :: [input] Name of period to add
 */
 void MuonFitPropertyBrowser::addPeriodCheckbox(const QString &name) {
+  // check period is sensible
+  // no frational periods
+  if (name.contains(".")) {
+    return;
+  } else {
+    auto tmp = name.toStdString();
+    std::vector<std::string> numbers;
+    auto num = tmp.find(",");
+    while (num != std::string::npos) {
+      numbers.push_back(tmp.substr(0, num));
+      tmp = tmp.substr(num + 1);
+      num = tmp.find(",");
+    }
+    numbers.push_back(tmp);
+    // loop over results
+    for (auto value : numbers) {
+      auto tmp = value.find("-");
+      if (tmp != std::string::npos) {
+        // find a minus sign
+        auto before = value.substr(0, tmp);
+        auto after = value.substr(tmp + 1);
+
+      } else {
+        try {
+          boost::lexical_cast<int>(value);
+          if (m_periodBoxes.find(QString::fromStdString(value)) ==
+                  m_periodBoxes.end() &&
+              numbers.size() > 1) {
+            // if the box does not exist and there is more than 1 period in name
+            return;
+          }
+        } catch (boost::bad_lexical_cast) {
+          // none int value
+          return;
+        }
+      }
+    }
+  }
+
```

Comment:
I think it would be better to move all this validation into a separate method that returns a boolean.